index.bs

```diff
@@ -353,13 +367,17 @@ The <dfn method for=FileSystemFileHandle>createWritable(|options|)</dfn> method
   1. If |access| is not "{{PermissionState/granted}}",
      [=reject=] |result| with a "{{NotAllowedError}}" {{DOMException}} and abort.
```

Comment:
This needs a task.

Reply:
Just to clarify, you're suggesting that the promise rejection should happen on some task source, not the `[=file system entry/request access=]` algorithm - correct?

This pattern (rejecting a promise from a step run in parallel) is repeated many times throughout this file. Would it be alright to fix these in a follow-up?

Also, just to confirm, is the promise rejection in the `lockResultAlgorithm` steps below fine as-is because it's already running on the storage task source? Or does that also need to queue a task?

Side note: while looking at this, I realized that `[=a new promise=]` is supposed to take a Realm. This appears to only be done correctly [in one method](https://fs.spec.whatwg.org/#dom-filesystemhandle-issameentry) for this spec. I can also fix this in a follow-up...

Reply:
> This pattern (rejecting a promise from a step run in parallel) is repeated many times throughout this file. Would it be alright to fix these in a follow-up?

For an idea of what this could look like: https://github.com/whatwg/fs/pull/9/files#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985R273-R276

Reply:
Yeah, I guess a follow-up is okay.

And yeah, if you already have a task you're not in parallel and you can resolve promises as you wish. (Though sometimes you might still want to queue a task, it really depends on what kind of ordering you are after.)

Reply:
Actually, this used to be synchronous but now it runs in parallel, so I think we need to change this. If something was already in parallel and makes this mistake I think it's fine to leave it for a follow-up.

Reply:
> And yeah, if you already have a task you're not in parallel and you can resolve promises as you wish. (Though sometimes you might still want to queue a task, it really depends on what kind of ordering you are after.)

Awesome, thanks for the clarification!

> Actually, this used to be synchronous but now it runs in parallel, so I think we need to change this. If something was already in parallel and makes this mistake I think it's fine to leave it for a follow-up.

SGTM. Let me know if the pattern I sketched out in https://github.com/whatwg/fs/pull/9#discussion_r1107406159 is appropriate here

Reply:
This now runs in the `file system queue`